insights/client/data_collector.py

```diff
@@ -109,6 +110,11 @@ def _parse_command_spec(self, spec, precmds):
             precmd_alias = spec['pre_command']
             try:
                 precmd = precmds[precmd_alias]
+
+                if set.intersection(set(shlex.split(precmd)),
+                                    constants.command_blacklist):
+                    raise RuntimeError("Command Blacklist")
```

Comment:
@gravitypriest would you want to add the intersection set as part of the exception message?

Reply:
Sure, that seems reasonable. I'll update.